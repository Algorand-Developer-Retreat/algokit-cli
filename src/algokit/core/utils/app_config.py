from pathlib import Path

import click

AppName = "algokit"


def get_config_dir():
    return Path(click.get_app_dir(AppName))


def read(file_name: str):
    return get_config_dir().joinpath(file_name).read_text("utf-8")


def write(file_name: str, content: str):
    get_config_dir().mkdir(parents=True, exist_ok=True)
    config_file = get_config_dir().joinpath(file_name)
    click.echo(click.style(f"Writing content to {config_file}", fg="blue"))
    config_file.write_text(content, "utf-8")
