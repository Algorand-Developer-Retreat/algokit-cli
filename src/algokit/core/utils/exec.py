import subprocess
from pathlib import Path

import click


def run(
    command: str,
    working_directory: Path | None = None,
    suppress_output: bool = False,
    throw_on_error: str | None = None,
):
    click.echo(click.style(f"Running '{command}' in '{working_directory or Path.cwd()}'", fg="blue"))
    result = subprocess.run(command.split(), stdout=subprocess.PIPE, stderr=subprocess.PIPE, cwd=working_directory)
    if not suppress_output and result.stdout.decode("utf-8").strip().__len__() > 0:
        click.echo(click.style(result.stdout.decode("utf-8"), fg="blue"))

    if throw_on_error is not None and result.returncode != 0:
        click.echo(click.style(result.stderr.decode("utf-8"), fg="red"))
        raise click.ClickException(throw_on_error)

    return result
