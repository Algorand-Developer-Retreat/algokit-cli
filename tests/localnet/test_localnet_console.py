from subprocess import CompletedProcess

from pytest_mock import MockerFixture
from utils.approvals import verify
from utils.click_invoker import invoke
from utils.proc_mock import ProcMock


def test_goal_console(proc_mock: ProcMock, mocker: MockerFixture):
    mocker.patch("algokit.core.proc.subprocess_run").return_value = CompletedProcess(
        ["docker", "exec"], 0, "STDOUT+STDERR"
    )

    result = invoke("localnet console")

    assert result.exit_code == 0
    verify(result.output)
