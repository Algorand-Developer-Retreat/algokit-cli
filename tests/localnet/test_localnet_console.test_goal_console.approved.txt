DEBUG: Running 'docker compose version --format json' in '{current_working_directory}'
DEBUG: docker: {"version": "v2.5.0"}
DEBUG: Running 'docker version' in '{current_working_directory}'
DEBUG: docker: STDOUT
DEBUG: docker: STDERR
DEBUG: Running 'docker version' in '{current_working_directory}'
DEBUG: docker: STDOUT
DEBUG: docker: STDERR
Opening Bash console on the algod node; execute `exit` to return to original console
DEBUG: Running 'docker exec -it -w /root algokit_algod bash' in '{current_working_directory}'
